services/QuillLMS/app/services/classroom_units_saver.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+class ClassroomUnitsSaver < ApplicationService
+  attr_reader :classrooms_data, :concatenate_existing_student_ids, :new_classroom_units_data, :unit_id
+
+  def initialize(classrooms_data, concatenate_existing_student_ids, unit_id)
+    @concatenate_existing_student_ids = concatenate_existing_student_ids
+    @new_classroom_units_data = []
+    @unit_id = unit_id
+
+    @classrooms_data =
+      classrooms_data
+        .map(&:symbolize_keys)
+        .select { |classroom_data| classroom_data.key?(:id) }
```

Comment:
take or leave, but since `nil` is `falsey` the select/reject lines could just be 
```ruby
.select { |classroom_data| classroom_data[:id] }